File: leetcode/26/26.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if nums == []:
            return 0
        
        tail = 0
        leng = len(nums)
        for i in range(1, leng):
            if nums[i] != nums[tail]:
                tail += 1
                nums[tail] = nums[i]
        return tail + 1

File: leetcode/26/26.cc
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 1;
        while(i < nums.size()){
            if(nums[i-1] == nums[i]){
                nums.erase(nums.begin()+i-1);
            } else {
                i++;
            }
        }
        return nums.size();;
    }
};